From 31d9656fc1594bc87a7285e9e30f1e9556d038aa Mon Sep 17 00:00:00 2001
From: Florian Zaruba <zarubaf@iis.ee.ethz.ch>
Date: Mon, 23 Aug 2021 16:19:27 +0200
Subject: [PATCH] occamy: Chip simulation fixes

---
 src/util/instr_tracer.sv | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/util/instr_tracer.sv b/src/util/instr_tracer.sv
index 2fbe781..6d9c860 100644
--- a/src/util/instr_tracer.sv
+++ b/src/util/instr_tracer.sv
@@ -140,10 +140,12 @@ module instr_tracer (
       // --------------
       // Exceptions
       // --------------
+      `ifndef VCS
       if (tracer_if.pck.exception.valid && !(tracer_if.pck.debug_mode && tracer_if.pck.exception.cause == riscv::BREAKPOINT)) begin
         // print exception
         printException(tracer_if.pck.commit_instr[0].pc, tracer_if.pck.exception.cause, tracer_if.pck.exception.tval);
       end
+      `endif
       // ----------------------
       // Commit Registers
       // ----------------------
-- 
2.25.1.377.g2d2118b814

