From 619335b9f9b01a63d5f6a391d6bb0e152fe9b28a Mon Sep 17 00:00:00 2001
From: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
Date: Mon, 6 Sep 2021 14:46:52 +0200
Subject: [PATCH 9/9] ariane: Mark WB's AXI requests as modifiable

Signed-off-by: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
---
 src/axi_adapter.sv | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/axi_adapter.sv b/src/axi_adapter.sv
index 6d04839..e2504b7 100644
--- a/src/axi_adapter.sv
+++ b/src/axi_adapter.sv
@@ -76,7 +76,7 @@ module axi_adapter #(
     axi_req_o.aw.size   = {1'b0, size_i}; // 1, 2, 4 or 8 bytes
     axi_req_o.aw.burst  = axi_pkg::BURST_INCR; // Use BURST_INCR for AXI regular transaction
     axi_req_o.aw.lock   = 1'b0;
-    axi_req_o.aw.cache  = 4'b0;
+    axi_req_o.aw.cache  = axi_pkg::CACHE_MODIFIABLE;
     axi_req_o.aw.qos    = 4'b0;
     axi_req_o.aw.id     = id_i;
     axi_req_o.aw.atop   = '0; // currently not used
@@ -92,7 +92,7 @@ module axi_adapter #(
     axi_req_o.ar.size   = {1'b0, size_i}; // 1, 2, 4 or 8 bytes
     axi_req_o.ar.burst  = (CRITICAL_WORD_FIRST ? axi_pkg::BURST_WRAP : axi_pkg::BURST_INCR); // wrapping transfer in case of a critical word first strategy
     axi_req_o.ar.lock   = 1'b0;
-    axi_req_o.ar.cache  = 4'b0;
+    axi_req_o.ar.cache  = axi_pkg::CACHE_MODIFIABLE;
     axi_req_o.ar.qos    = 4'b0;
     axi_req_o.ar.id     = id_i;
     axi_req_o.ar.user   = '0;
-- 
2.16.5

