From 72c3356f02925c2b764c4a246707f7d1220573bf Mon Sep 17 00:00:00 2001
From: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
Date: Tue, 21 Mar 2023 16:26:30 +0100
Subject: [PATCH] cva6: Expose sram cfg ports

Signed-off-by: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
---
 core/ariane.sv | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/core/ariane.sv b/core/ariane.sv
index c724ad9d..7644fddb 100644
--- a/core/ariane.sv
+++ b/core/ariane.sv
@@ -32,7 +32,12 @@ module ariane import ariane_pkg::*; #(
   // Core ID, Cluster ID and boot address are considered more or less static
   input  logic [riscv::VLEN-1:0]       boot_addr_i,  // reset boot address
   input  logic [riscv::XLEN-1:0]       hart_id_i,    // hart id in a multicore environment (reflected in a CSR)
-
+  // SRAM config
+  input sram_cfg_t                     sram_cfg_idata_i,
+  input sram_cfg_t                     sram_cfg_itag_i,
+  input sram_cfg_t                     sram_cfg_ddata_i,
+  input sram_cfg_t                     sram_cfg_dtag_i,
+  input sram_cfg_t                     sram_cfg_dvalid_dirty_i,
   // Interrupt inputs
   input  logic [1:0]                   irq_i,        // level sensitive IR lines, mip & sip (async)
   input  logic                         ipi_i,        // inter-processor interrupts (async)
@@ -79,6 +84,11 @@ module ariane import ariane_pkg::*; #(
     .rst_ni               ( rst_ni                    ),
     .boot_addr_i          ( boot_addr_i               ),
     .hart_id_i            ( hart_id_i                 ),
+    .sram_cfg_idata_i     ( sram_cfg_idata_i          ),
+    .sram_cfg_itag_i      ( sram_cfg_itag_i           ),
+    .sram_cfg_ddata_i     ( sram_cfg_ddata_i          ),
+    .sram_cfg_dtag_i      ( sram_cfg_dtag_i           ),
+    .sram_cfg_dvalid_dirty_i ( sram_cfg_dvalid_dirty_i ),
     .irq_i                ( irq_i                     ),
     .ipi_i                ( ipi_i                     ),
     .time_irq_i           ( time_irq_i                ),
-- 
2.16.5

