From 3dbbef482c0ca6958ea6ec47e568aaf971c036d2 Mon Sep 17 00:00:00 2001
From: Luca Colagrande <bigcola.96@gmail.com>
Date: Sat, 27 Nov 2021 11:53:48 +0100
Subject: [PATCH] Mask non-enabled interrupts in WFI control

---
 src/csr_regfile.sv | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/csr_regfile.sv b/src/csr_regfile.sv
index e7f89f8..e9b9d6a 100644
--- a/src/csr_regfile.sv
+++ b/src/csr_regfile.sv
@@ -980,9 +980,9 @@ module csr_regfile import ariane_pkg::*; #(
     always_comb begin : wfi_ctrl
         // wait for interrupt register
         wfi_d = wfi_q;
-        // if there is any interrupt pending un-stall the core
+        // if there is any (enabled) interrupt pending un-stall the core
         // also un-stall if we want to enter debug mode
-        if (|mip_q || debug_req_i || irq_i[1]) begin
+        if (|(mip_q & mie_q) || debug_req_i || irq_i[1]) begin
             wfi_d = 1'b0;
         // or alternatively if there is no exception pending and we are not in debug mode wait here
         // for the interrupt
-- 
2.28.0

