From 097522a7c17011156b55a3aab4139a82d2eb0760 Mon Sep 17 00:00:00 2001
From: Paul Scheffler <paulsc@iis.ee.ethz.ch>
Date: Mon, 25 Apr 2022 17:42:16 +0200
Subject: [PATCH] spi_host: More fixes (#390, #393)

---
 rtl/spi_host_fsm.sv | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rtl/spi_host_fsm.sv b/rtl/spi_host_fsm.sv
index 9c28a41..3e01bff 100644
--- a/rtl/spi_host_fsm.sv
+++ b/rtl/spi_host_fsm.sv
@@ -291,7 +291,7 @@ module spi_host_fsm
           if (idle_cntr_q == 4'h0) begin
             prestall_st_d = WaitLead;
           end else begin
-            prestall_st_d = WaitIdle;
+            prestall_st_d = CSBSwitch;
           end
         end
         IdleCSBActive: begin
@@ -483,7 +483,7 @@ module spi_host_fsm
 
   assign wr_en_internal    = byte_starting & cmd_wr_en;
   assign shift_en_internal = bit_shifting;
-  assign rd_en_internal    = byte_ending & cmd_rd_en;
+  assign rd_en_internal    = byte_ending & cmd_rd_en_q;
   assign speed_o           = cmd_speed;
   assign sample_en_d       = byte_starting | shift_en_o;
   assign full_cyc_o        = full_cyc;
-- 
2.16.5

