From 55a60a40826e2b60f2cf2af5a3d5ef3470881570 Mon Sep 17 00:00:00 2001
From: Noah Huetter <noahhuetter@gmail.com>
Date: Tue, 12 Apr 2022 12:09:30 +0200
Subject: riscv-dbg: Re-align dmi_bscane_tap interface with dmi_jtag_tap

Signed-off-by: Noah Huetter <noahhuetter@gmail.com>
---
 src/dmi_bscane_tap.sv | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/dmi_bscane_tap.sv b/src/dmi_bscane_tap.sv
index b1c12a85..5fe5d86e 100644
--- a/src/dmi_bscane_tap.sv
+++ b/src/dmi_bscane_tap.sv
@@ -32,7 +32,7 @@ module dmi_jtag_tap #(
   input  logic        testmode_i,
 
   output logic        tck_o,
-  output logic        trst_no,
+  output logic        dmi_clear_o,
   output logic        update_o,
   output logic        capture_o,
   output logic        shift_o,
@@ -44,16 +44,12 @@ module dmi_jtag_tap #(
   input  logic        dmi_tdo_i
 );
 
-  /// DTMCS Register
-  logic trst;
-  assign trst_no = ~trst;
-
   BSCANE2 #(
     .JTAG_CHAIN (3)
   ) i_tap_dtmcs (
     .CAPTURE (capture_o),
     .DRCK (),
-    .RESET (trst),
+    .RESET (dmi_clear_o),
     .RUNTEST (),
     .SEL (dtmcs_select_o),
     .SHIFT (shift_o),
-- 
2.16.5

