From 1a4e8618db956c8d5b6d7333ff70c05a53433b07 Mon Sep 17 00:00:00 2001
From: Florian Zaruba <zarubaf@iis.ee.ethz.ch>
Date: Thu, 26 Nov 2020 18:51:44 +0100
Subject: [PATCH] axi-lite-xbar: Add missing cache ports

---
 src/axi_lite_xbar.sv | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/axi_lite_xbar.sv b/src/axi_lite_xbar.sv
index a11389b..61fed01 100644
--- a/src/axi_lite_xbar.sv
+++ b/src/axi_lite_xbar.sv
@@ -173,6 +173,8 @@ module axi_lite_xbar #(
     ) i_dec_err_conv (
       .slv_req_lite_i  ( slv_reqs[i][Cfg.NoMstPorts]  ),
       .slv_resp_lite_o ( slv_resps[i][Cfg.NoMstPorts] ),
+      .slv_aw_cache_i  ( 4'd0                         ),
+      .slv_ar_cache_i  ( 4'd0                         ),
       .mst_req_o       ( decerr_req                   ),
       .mst_resp_i      ( decerr_resp                  )
     );
-- 
2.24.3 (Apple Git-128)

