From 8d4bcc7b6cb7f91394e3fa96dc2b6628a114bd88 Mon Sep 17 00:00:00 2001
From: Florian Zaruba <zarubaf@iis.ee.ethz.ch>
Date: Wed, 13 Jan 2021 01:05:03 +0100
Subject: [PATCH 4/4] axi_test: Also optionally randomize responses

---
 src/axi_test.sv | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/axi_test.sv b/src/axi_test.sv
index 6d74b0e..60f4583 100644
--- a/src/axi_test.sv
+++ b/src/axi_test.sv
@@ -1259,6 +1259,7 @@ package axi_test;
     // Stimuli application and test time
     parameter time  TA = 0ps,
     parameter time  TT = 0ps,
+    parameter bit   RAND_RESP = 0,
     // Upper and lower bounds on wait cycles on Ax, W, and resp (R and B) channels
     parameter int   AX_MIN_WAIT_CYCLES = 0,
     parameter int   AX_MAX_WAIT_CYCLES = 100,
@@ -1332,6 +1333,8 @@ package axi_test;
         ar_beat = ar_queue.peek();
         rand_success = std::randomize(r_beat); assert(rand_success);
         r_beat.r_id = ar_beat.ax_id;
+        if (RAND_RESP && !ar_beat.ax_atop[5])
+          r_beat.r_resp[1] = $random();
         if (ar_beat.ax_lock)
           r_beat.r_resp[0]= $random();
         rand_wait(R_MIN_WAIT_CYCLES, R_MAX_WAIT_CYCLES);
@@ -1382,6 +1385,8 @@ package axi_test;
         aw_beat = aw_queue.pop_front();
         rand_success = std::randomize(b_beat); assert(rand_success);
         b_beat.b_id = aw_beat.ax_id;
+        if (RAND_RESP && !aw_beat.ax_atop[5])
+          b_beat.b_resp[1] = $random();
         if (aw_beat.ax_lock) begin
           b_beat.b_resp[0]= $random();
         end
-- 
2.25.1.377.g2d2118b814

