From 4ee117dc6ad2d979c073612a7f175a96cad3d868 Mon Sep 17 00:00:00 2001
From: Florian Zaruba <zarubaf@iis.ee.ethz.ch>
Date: Mon, 18 Jan 2021 14:51:56 +0100
Subject: [PATCH] axi: Add `typedef_all`

---
 include/axi/typedef.svh | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/axi/typedef.svh b/include/axi/typedef.svh
index 9c9ade6..a476ee8 100644
--- a/include/axi/typedef.svh
+++ b/include/axi/typedef.svh
@@ -101,6 +101,14 @@
   } resp_t;
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+`define AXI_TYPEDEF_ALL(__name, __addr_t, __id_t, __data_t, __strb_t, __user_t)                 \
+  `AXI_TYPEDEF_AW_CHAN_T(__name``_aw_chan_t, __addr_t, __id_t, __user_t)                        \
+  `AXI_TYPEDEF_W_CHAN_T(__name``_w_chan_t, __data_t, __strb_t, __user_t)                        \
+  `AXI_TYPEDEF_B_CHAN_T(__name``_b_chan_t, __id_t, __user_t)                                    \
+  `AXI_TYPEDEF_AR_CHAN_T(__name``_ar_chan_t, __addr_t, __id_t, __user_t)                        \
+  `AXI_TYPEDEF_R_CHAN_T(__name``_r_chan_t, __data_t, __id_t, __user_t)                          \
+  `AXI_TYPEDEF_REQ_T(__name``_req_t, __name``_aw_chan_t, __name``_w_chan_t, __name``_ar_chan_t) \
+  `AXI_TYPEDEF_RESP_T(__name``_resp_t, __name``_b_chan_t, __name``_r_chan_t)
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // AXI-Lite (4+Prot) Channel and Request/Response Structs
-- 
2.25.1.377.g2d2118b814

