From 0de63a0fb63ac8a005da9948db583b257de76afb Mon Sep 17 00:00:00 2001
From: Gianna Paulin <pauling@student.ethz.ch>
Date: Mon, 25 Apr 2022 11:04:08 +0200
Subject: [PATCH] vendor: add AtopSupport parameter to axi_id_serialize

---
 src/axi_id_serialize.sv | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/axi_id_serialize.sv b/src/axi_id_serialize.sv
index 4e81a901..e212c8a7 100644
--- a/src/axi_id_serialize.sv
+++ b/src/axi_id_serialize.sv
@@ -27,6 +27,7 @@
 /// This module contains one [`axi_serializer`](module.axi_serializer) per master port ID (given by
 /// the `AxiMstPortMaxUniqIds parameter`).
 module axi_id_serialize #(
+  parameter bit          AtopSupport  = 1'b1,
   /// ID width of the AXI4+ATOP slave port
   parameter int unsigned AxiSlvPortIdWidth = 32'd0,
   /// Maximum number of transactions that can be in flight at the slave port.  Reads and writes are
@@ -150,6 +151,7 @@ module axi_id_serialize #(
   slv_resp_t [AxiMstPortMaxUniqIds-1:0] to_serializer_resps;
 
   axi_demux #(
+    .AtopSupport ( AtopSupport          ),
     .AxiIdWidth  ( AxiSlvPortIdWidth    ),
     .aw_chan_t   ( slv_aw_t             ),
     .w_chan_t    ( w_t                  ),
-- 
2.16.5

