From ca1335818b954db5c7eae69b52534aea31706270 Mon Sep 17 00:00:00 2001
From: Gianna Paulin <pauling@student.ethz.ch>
Date: Tue, 3 May 2022 10:05:47 +0200
Subject: [PATCH] add patch for PR: Fix axi_demux AW stall

---
 src/axi_demux.sv | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/axi_demux.sv b/src/axi_demux.sv
index 8124d126..1dab0075 100644
--- a/src/axi_demux.sv
+++ b/src/axi_demux.sv
@@ -273,10 +273,12 @@ module axi_demux #(
           atop_inject     = slv_aw_chan.atop[axi_pkg::ATOP_R_RESP] & AtopSupport;
         end
       end else begin
-        // Process can start handling a transaction if its `i_aw_id_counter` and `w_fifo` have
-        // space in them. Further check if we could inject something on the AR channel (only if
-        // ATOPs are supported).
-        if (!aw_id_cnt_full && !w_fifo_full && (!ar_id_cnt_full || !AtopSupport)) begin
+        // An AW can be handled if `i_aw_id_counter` and `i_w_fifo` are not full.  An ATOP that
+        // requires an R response can be handled if additionally `i_ar_id_counter` is not full (this
+        // only applies if ATOPs are supported at all).
+        if (!aw_id_cnt_full && !w_fifo_full &&
+            (!(ar_id_cnt_full && slv_aw_chan_select.aw_chan.atop[axi_pkg::ATOP_R_RESP]) ||
+             !AtopSupport)) begin
           // there is a valid AW vector make the id lookup and go further, if it passes
           if (slv_aw_valid && (!aw_select_occupied ||
              (slv_aw_select == lookup_aw_select))) begin
-- 
2.16.5

