From 9ad85fa35a8fe7f2199bddac69434e0f0e69ee38 Mon Sep 17 00:00:00 2001
From: Gianna Paulin <pauling@student.ethz.ch>
Date: Tue, 3 May 2022 10:45:54 +0200
Subject: [PATCH] axi: Compatibility axi_id_remap

---
 src/axi_id_remap.sv | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/src/axi_id_remap.sv b/src/axi_id_remap.sv
index 677bc14f..f424360b 100644
--- a/src/axi_id_remap.sv
+++ b/src/axi_id_remap.sv
@@ -284,13 +284,21 @@ module axi_id_remap #(
             aw_id_d = wr_push_oup_id;
           end
         end
-        priority casez ({mst_req_o.ar_valid, mst_resp_i.ar_ready,
-                         mst_req_o.aw_valid, mst_resp_i.aw_ready})
-          4'b1010: state_d = HoldAx;
-          4'b10??: state_d = HoldAR;
-          4'b??10: state_d = HoldAW;
-          default: state_d = Ready;
-        endcase
+        if ({mst_req_o.ar_valid, mst_resp_i.ar_ready,
+             mst_req_o.aw_valid, mst_resp_i.aw_ready} == 4'b1010) begin
+          state_d = HoldAx;
+        end else if ({mst_req_o.ar_valid, mst_resp_i.ar_ready} == 2'b10) begin
+          state_d = HoldAR;
+        end else if ({mst_req_o.aw_valid, mst_resp_i.aw_ready} == 2'b10) begin
+          state_d = HoldAW;
+        end else state_d = Ready;
+        // priority casez ({mst_req_o.ar_valid, mst_resp_i.ar_ready,
+        //                  mst_req_o.aw_valid, mst_resp_i.aw_ready})
+        //   4'b1010: state_d = HoldAx;
+        //   4'b10??: state_d = HoldAR;
+        //   4'b??10: state_d = HoldAW;
+        //   default: state_d = Ready;
+        // endcase
 
         if (mst_req_o.ar_valid && mst_resp_i.ar_ready) begin
           ar_prio_d = 1'b0; // Reset AR priority, because handshake was successful in this cycle.
-- 
2.16.5

