From f5e544ce8b8be4d259357b369965956e7380b051 Mon Sep 17 00:00:00 2001
From: Florian Zaruba <zarubaf@iis.ee.ethz.ch>
Date: Thu, 26 Nov 2020 17:36:13 +0100
Subject: [PATCH 2/2] tc_sram: Disable random mode in Verilator

---
 src/rtl/tc_sram.sv | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/rtl/tc_sram.sv b/src/rtl/tc_sram.sv
index 53530e0..0e9ff59 100644
--- a/src/rtl/tc_sram.sv
+++ b/src/rtl/tc_sram.sv
@@ -90,7 +90,9 @@ module tc_sram #(
         case (SimInit)
           "zeros":  init_val[i][j] = 1'b0;
           "ones":   init_val[i][j] = 1'b1;
+          `ifndef VERILATOR
           "random": init_val[i][j] = $urandom();
+          `endif
           default:  init_val[i][j] = 1'bx;
         endcase
       end
-- 
2.24.3 (Apple Git-128)

