From 959278d749d09bda7df305c94d29f4212d0f8117 Mon Sep 17 00:00:00 2001
From: Paul Scheffler <paulsc@iis.ee.ethz.ch>
Date: Wed, 9 Mar 2022 14:42:19 +0100
Subject: [PATCH] common_cells: Use glitch-free muxes in `rstgen_bypass`

---
 src/rstgen_bypass.sv | 34 ++++++++++++++++++++++------------
 1 file changed, 22 insertions(+), 12 deletions(-)

diff --git a/src/rstgen_bypass.sv b/src/rstgen_bypass.sv
index c51ee83..7d3bf57 100644
--- a/src/rstgen_bypass.sv
+++ b/src/rstgen_bypass.sv
@@ -27,18 +27,28 @@ module rstgen_bypass #(
     logic rst_n;
 
     logic [NumRegs-1:0] synch_regs_q;
-    // bypass mode
-    always_comb begin
-        if (test_mode_i == 1'b0) begin
-            rst_n   = rst_ni;
-            rst_no  = synch_regs_q[NumRegs-1];
-            init_no = synch_regs_q[NumRegs-1];
-        end else begin
-            rst_n   = rst_test_mode_ni;
-            rst_no  = rst_test_mode_ni;
-            init_no = 1'b1;
-        end
-    end
+
+    // bypass mode: use glitch-free (clock) multiplexers
+    tc_clk_mux2 i_tc_clk_mux2_rst_n (
+        .clk0_i     ( rst_ni ),
+        .clk1_i     ( rst_test_mode_ni ),
+        .clk_sel_i  ( test_mode_i ),
+        .clk_o      ( rst_n )
+    );
+
+    tc_clk_mux2 i_tc_clk_mux2_rst_no (
+        .clk0_i     ( synch_regs_q[NumRegs-1] ),
+        .clk1_i     ( rst_test_mode_ni ),
+        .clk_sel_i  ( test_mode_i ),
+        .clk_o      ( rst_no )
+    );
+
+    tc_clk_mux2 i_tc_clk_mux2_init_no (
+        .clk0_i     ( synch_regs_q[NumRegs-1] ),
+        .clk1_i     ( 1'b1 ),
+        .clk_sel_i  ( test_mode_i ),
+        .clk_o      ( init_no )
+    );
 
     always @(posedge clk_i or negedge rst_n) begin
         if (~rst_n) begin
-- 
2.16.5

