From 71457e7936957e345780a1c28af835c62ad98fdd Mon Sep 17 00:00:00 2001
From: Samuel Riedel <sriedel@iis.ee.ethz.ch>
Date: Tue, 3 May 2022 18:45:27 +0200
Subject: [PATCH] axi_riscv_atomics: Remove timeunit in testbench

---
 test/axi_riscv_lrsc_tb.sv | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/test/axi_riscv_lrsc_tb.sv b/test/axi_riscv_lrsc_tb.sv
index 9abbea54..937ef171 100644
--- a/test/axi_riscv_lrsc_tb.sv
+++ b/test/axi_riscv_lrsc_tb.sv
@@ -31,9 +31,6 @@ module axi_riscv_lrsc_tb #(
     parameter bit VERBOSE = 1'b0
 );
 
-    timeunit 1ns;
-    timeprecision 10ps;
-
     localparam time TCLK = 10ns;
     localparam time TA = TCLK * 1/4;
     localparam time TT = TCLK * 3/4;
-- 
2.16.5

